src/sotahttpclient.cc

```diff
@@ -85,7 +84,7 @@ void SotaHttpClient::startInstall(const data::UpdateRequestId &InstallingUpdate)
 void SotaHttpClient::sendUpdateReport(data::UpdateReport update_report) {
   std::string url = core_url + "/mydevice/" + config.device.uuid;
   url += "/updates/" + update_report.update_id;
-  http->post(url, update_report.toJson()["operation_results"]);
+  http->post(url, Json::FastWriter().write(update_report.toJson()["operation_results"]));
```

Comment:
Looking at our code base, the only thing we send via http->post and http->put is JSON.  What about putting the JSON serialization inside the HttpClient?